src/ConnectedMachine/readme.md

```diff
@@ -40,45 +40,108 @@ In this directory, run AutoRest:
 > see https://aka.ms/autorest
 
 ``` yaml
-powershell: true
-azure: true
-branch: master
-repo: https://github.com/Azure/azure-rest-api-specs/tree/$(branch)
-metadata:
-  authors: Microsoft Corporation
-  owners: Microsoft Corporation
-  description: 'Microsoft Azure PowerShell: $(service-name) cmdlets'
-  copyright: Microsoft Corporation. All rights reserved.
-  tags: Azure ResourceManager ARM PSModule $(service-name)
-  companyName: Microsoft Corporation
-  requireLicenseAcceptance: true
-  licenseUri: https://aka.ms/azps-license
-  projectUri: https://github.com/Azure/azure-powershell
-prefix: Az
-subject-prefix: 'Connected'
-module-name: $(prefix).$(service-name)
-namespace: Microsoft.Azure.PowerShell.Cmdlets.$(service-name)
-clear-output-folder: true
-output-folder: .
-input-file:
-  - $(repo)/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2019-08-02/HybridCompute.json
-
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
 module-version: 0.1.0
 title: ConnectedMachine
+subject-prefix: 'Connected'
+input-file:
+  - $(repo)/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/stable/2019-12-12/HybridCompute.json
 
 directive:
   - where:
       subject: Operation
     hide: true
   - where: $.definitions.Identifier.properties
     suppress: R3019
-  - remove-operation: 
+
+  # GetViaIdentity isn't useful until Azure PowerShell supports piping of different subjects
+  - where:
+      verb: New
+      variant: ^GetViaIdentity\d?$
+    remove: true
+
+  # Set correct variants for PUT and PATCH verbs
+  - where:
+      verb: New
+      variant: ^CreateViaIdentity\d?$|^Create\d?$
+    remove: true
+  - where:
+      verb: Set
+      variant: ^Update\d?$|^UpdateViaIdentity\d?$
+    remove: true
+  - where:
+      verb: Update
+      variant: ^Update\d?$|^UpdateViaIdentity\d?$
+    remove: true
+
+  # New, Update, and Set all do the same thing so we hide New and Update.
+  - where:
+      verb: Update|New
+      subject: MachineExtension
+    hide: true
```

Comment:
The guideline to generated modules requires `New-` and `Update-` instead of `Set-`. See https://github.com/Azure/azure-powershell/wiki/How-to-On-Board-New-RP-with-Azure-PowerShell-Generator#service-readiness

Reply:
As of piping from connected machine to extension, the only way is through customization.

Reply:
@isra-fel I did have a `New-` and `Update-` at one point but the generated `New-` cmdlet doesn't error if the extension already exists and the `Update-` also created new extensions if they didn't already exist. Is that ok?

I also only have a `Set-` because that's what the Compute cmdlets have (they only have a `Set-` and these extension cmdlets are the same swagger definition as the ones for Compute)

Reply:
- In the case of  `New-` an existing resource, if backend service supports `Etag` powershell will error out. Besides, it's not a common scenario and impact is limited.
- `Update-` calls `PATCH` API under the hood. It shouldn't create the resource if not already exist. It's service's bug if it does.
- Generated modules and hand-written modules have different guidelines. You don't have to align very much to old cmdlets.

Reply:
Thanks @isra-fel, I will unhide these cmdlets.

@edyoung @raghushantha this might be a couple issues on the service side (lack of `ETag` and not handling `PATCH` correctly). Can you help here?

Reply:
I gave this another go and `Update-AzConnectedMachineExtension` is now working as expected. It wasn't before but now it is.

Since @isra-fel mentioned that `New-AzConnectedMachineExtension` not erroring if the resource already exists is not a common scenario and impact is limited,

this thread isn't that high of priority.

Eventually extensions could support `ETag` but it's not urgent.